### ISSUE TITLE
Interesting blog post about creating and publishing modules

### ISSUE THREAD

**Issue by subotic** [dhlab-basel/salsah-modules#7]
https://medium.com/@cyrilletuzi/how-to-build-and-publish-an-angular-module-7ad19c0b4464

**Comment by subotic** [comment]
```
Finally, respect the Angular rule: never use browser-specific APIs (like the DOM) directly. If you do so, your module won’t be compatible with Universal server rendering and other Angular advanced options. If you really need to use browser-specific APIs (localStorage…), you should try/catch errors.
```

Hmm, we use `localStorage` in the `AuthenticationService`

**Comment by kilchenmann** [comment]
good point, but bad for us ... 🙈

Btw. I followed this post: https://medium.com/@nikolasleblanc/building-an-angular-4-component-library-with-the-angular-cli-and-ng-packagr-53b2ade0701e with the example project for ng-packagr called [ng-packaged](https://github.com/dherges/ng-packaged)

**Comment by subotic** [comment]
> good point, but bad for us ...

Actually, I think this is very good news. I never liked the idea of simply writing to localStorage or sessionStorage from everywhere. This can become (and it probably is already) very messy and confusing. It is just a matter of time until we start to write over each other's variables unintentionally.

I'm sure that we are not the first application which needs some kind of centralized state management. We should look into this.

**Comment by kilchenmann** [comment]
Yes of course, the localStorage solution is a quick "hack" which shouldn't be used in production mode anymore. In the salsah-modules, which will be implemented into the salsah-app later, we don't use the local and session storage anymore.
